src/main/java/org/testng/SuiteRunner.java

```diff
@@ -66,6 +66,7 @@
   private ITestObjectFactory objectFactory;
   private Boolean skipFailedInvocationCounts = Boolean.FALSE;
   private List<IReporter> reporters = Lists.newArrayList();
+  private List<IReporter2> reporters2 = Lists.newArrayList();
```

Comment:
In fact, you don't need 2 lists. You can manage `IReporter` with a very simple IReporter2 adapter.